viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    movies_info = {}
+    movies_info['title'] = title
+    movies_info['genre'] = genre
+    movies_info['rating'] = rating
+    if not title or not genre or not rating:
+        return None
+    return movies_info
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:   
+            user_data['watchlist'].remove(movie)
+            user_data['watched'].append(movie)
+            # return user_data
+    # If the movie is not found in the watchlist or the watchlist is empty, return the original user_data
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data.get("watched", [])
+    if not watched_movies:
+        return 0.0
+    ratings_sum = 0.0 
+    for movie_dict in watched_movies:
+        ratings_sum += movie_dict.get('rating', 0.0)  
+    return ratings_sum / len(watched_movies)
+
+def get_most_watched_genre(user_data):
+    genre_occur = {}
+    if not user_data["watched"]:
+        return None
+    for movie_dict in user_data["watched"]:
+        genre_occur[movie_dict["genre"]] = genre_occur.get(movie_dict["genre"], 0) + 1
+    return max(genre_occur, key=genre_occur.get)
+
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data): 

```

Comment:
[perf] I think this function is optimal, from the perspective of big-O time complexity. I wouldn't change anything, personally, but it's always worth looking at it and asking whether you like the performance tradeoff compared to a simpler-to-read but less optimal implementation. 😊